src/patients/patients.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { PatientsRepository } from 'patients/patients.repository';
+import { PatientEntity } from 'patients/patient.entity';
+import { PatientCreateDto, PatientDto } from 'patients/dtos/patient.dto';
+import { JwtPayload } from 'common/jwt/jwt.strategy';
+
+@Injectable()
+export class PatientsService {
+  constructor(
+    @InjectRepository(PatientsRepository)
+    private patientsRepository: PatientsRepository,
+  ) {}
+  async getPatient(user: JwtPayload): Promise<PatientEntity> {
+    const found = await this.patientsRepository.findOne({
+      where: { userId: user.id },
+    });
+
+    if (found) {
+      return found;
+    }
+    throw new NotFoundException();
+  }
+
+  async getPatientById(id: string): Promise<PatientEntity> {
+    const found = await this.patientsRepository.findOne({
+      where: { id },
+    });
+
+    if (found) {
+      return found;
+    }
+    throw new NotFoundException();
+  }
+
+  async createPatient(profile: PatientCreateDto): Promise<PatientEntity> {
+    const entity = await this.patientsRepository.createPatient(profile);
+
+    if (entity) {
+      return entity;
+    }
+    throw new NotFoundException();
+  }
+
+  async updatePatient(id: string, dto: PatientDto): Promise<PatientEntity> {
+    const entity = await this.getPatientById(id);
+    const updated = { ...entity, ...dto };
+
+    await this.patientsRepository.save(updated);
```

Comment:
you are performing two separate requests here, isn't there a more efficient way with an update query?